File: Commitji.Core/Model/SemVerChange.fs
namespace Commitji.Core.Model

type SemVerChange =
    | Major
    | Minor
    | Patch

    member this.Code =
        match this with
        | Major -> "Major"
        | Minor -> "Minor"
        | Patch -> "Patch"

[<RequireQualifiedAccess>]
module SemVerChange =
    let determine breakingChange prefix =
        match breakingChange, prefix with
        | BreakingChange.Yes, _ -> Some SemVerChange.Major
        | BreakingChange.No, Prefix.Fix -> Some SemVerChange.Minor
        | BreakingChange.No, Prefix.Feat -> Some SemVerChange.Patch
        | BreakingChange.No, _ -> None

File: Commitji/Model/SemVerChange.cs
namespace Commitji.Model;

public record SemVerChange(string Name, string Format, bool BreakingChange, Prefix? Prefix)
{
    public static readonly IReadOnlyList<SemVerChange> All = new List<SemVerChange>
    {
        new("Major", "+1._._", true, null),
        new("Minor", "_.+1._", false, Prefix.Feat),
        new("Patch", "_._.+1", false, Prefix.Fix),
        new("None", "", false, null)
    };
}